openhands/controller/agent_controller.py

```diff
@@ -334,6 +334,10 @@ async def _handle_message_action(self, action: MessageAction) -> None:
     def _reset(self) -> None:
         """Resets the agent controller"""
 
+        # if the pending action has a tool call, reset the tool call
+        if self._pending_action and hasattr(self._pending_action, 'tool_call_metadata'):
+            self._pending_action.tool_call_metadata = None
```

Comment:
Any idea why we want to do this? Why not directly remove the pending_action?

Reply:
The failed tool calls are gathered in the [pending queue](https://github.com/All-Hands-AI/OpenHands/blob/409d132747bfcc9d77b5eb56a8936856a89b6df8/openhands/agenthub/codeact_agent/codeact_agent.py#L456) in the agent otherwise, and never get sent to the LLM. They are still in the stream, but the agent _never_ sees them, even though they appear real to the user (and to us, when sent in trajectories/logs etc).

Actually, I think we should give it an error observation. It appears to work, I'm still testing that. What do you think?

Reply:
What does "failed tool calls" means? Is it a tool call that somehow never get executed?

Reply:
Ah, yes, sorry. When there's a runtime error like one raised [here](https://github.com/All-Hands-AI/OpenHands/blob/8488dd2a03db024ca6f39f0f30135b1156f4bf2c/openhands/runtime/impl/eventstream/eventstream_runtime.py#L442), it's possible it never gets an observation. Like this: 
- https://github.com/All-Hands-AI/OpenHands/issues/5601
- https://github.com/All-Hands-AI/OpenHands/issues/5602

Reply:
ahh - then how would remove the tool call metadata here help though?

Reply:
Oh, I think you're right to question it - it should fail some asserts in the agent if it was removed. No, this is no good.

Reply:
Here in `reset`, we still have the `self._pending_action`, and we know this action isn't actually going to finish normally. We can try to create on spot an error obs, linked to that action? 🤔 

I'm not sure there are many alternatives right now, I can think of others but they won't work... Does the agent really need to support multiple pending tool calls, when in fact we're processing them by response (ModelResponse)? And we [override](https://github.com/All-Hands-AI/OpenHands/blob/8488dd2a03db024ca6f39f0f30135b1156f4bf2c/openhands/agenthub/codeact_agent/codeact_agent.py#L172) the tool call to have one per ModelResponse. We're executing it, it doesn't actually wait for another?

Reply:
>  We can try to create on spot an error obs, linked to that action? 🤔

I think this is probably a better way! This also preserve enough info so we can look into details when it happens again